java-app-best-practices: Add readOnlyPropagatesToServer (#22321)
pingcap/docs: best-practices/java-app-best-practices.md

- Go to TiDB monitoring dashboard and view the request command type through **Query Summary** > **CPS By Instance**.
- If the number of `COM_STMT_EXECUTE` in the request is far more than the number of `COM_STMT_PREPARE`, it means this setting already takes effect.

#### `readOnlyPropagatesToServer`

> You should disable the `readOnlyPropagatesToServer` property. This prevents the JDBC driver from sending the `SET SESSION TRANSACTION READ ONLY` statement to the server. TiDB does not support this statement, and it is also unnecessary because all TiDB nodes can handle read/write connections.

Reviewer: ```suggestion
Disable the readOnlyPropagatesToServer property. When this property is enabled, the JDBC driver sends the `SET SESSION TRANSACTION READ ONLY` statement to the server. TiDB does not support this statement, and it is unnecessary because all TiDB nodes accept both read and write connections.
```
